### PR TITLE
feat: introduce ScramInterruptedException to check PBKDF2 interruptions

### PR DESCRIPTION
**Pull request by jorsol**

### CODE REVIEW

File: scram-common/src/test/java/com/ongres/scram/common/HiFunctionTest.java [hunk 1]
```diff
@@ -37,11 +38,13 @@ class HiFunctionTest {
...
   // Character sets for different edge cases
-  private static final String ALPHANUMERIC = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+  private static final String ALPHANUMERIC =
+      "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
   private static final String SPECIAL = "!@#$%^&*()-_=+[{]}\\|;:'\",.<>/?`~ ";
   private static final String UNICODE_EMOJI = "🚀🔥🛡️🔑";
   private static final String UNICODE_EXTENDED = "こんにちは世界"; // "Hello World" in Japanese
-  private static final String COMBINED_POOL = ALPHANUMERIC + SPECIAL + UNICODE_EXTENDED + UNICODE_EMOJI;
+  private static final String COMBINED_POOL =
+      ALPHANUMERIC + SPECIAL + UNICODE_EXTENDED + UNICODE_EMOJI;
```
Comment (sehrope) [inline scram-common/src/test/java/com/ongres/scram/common/HiFunctionTest.java:47]: These haven't changed but maybe your auto formatter has?

File: scram-common/src/main/java/com/ongres/scram/common/CryptoUtil.java [hunk 2]
```diff
@@ -88,16 +100,20 @@ static byte[] hi(Mac mac, char[] password, byte[] salt, int iterationCount) {
...
+    boolean success = false;
 
     try {
       for (int i = 2; i <= iterationCount; i++) {
-        if ((i & (INTERRUPT_CHECK_STRIDE - 1)) == 0 && Thread.interrupted()) {
-          Thread.currentThread().interrupt();
-          Arrays.fill(result, (byte) 0);
-          throw new ScramRuntimeException("PBKDF2 computation was interrupted");
+        if ((i & (INTERRUPT_CHECK_STRIDE - 1)) == 0 && Thread.currentThread().isInterrupted()) {
+          throw new ScramInterruptedException("PBKDF2 computation interrupted at iteration " + i);
```
Comment (sehrope) [inline scram-common/src/main/java/com/ongres/scram/common/CryptoUtil.java:116]: Nice.

File: scram-common/src/test/java/com/ongres/scram/common/HiFunctionTest.java [hunk 3]
```diff
@@ -124,10 +127,10 @@ void testInterruptionBehavior() throws Exception {
     try {
       Thread.currentThread().interrupt();
 
-      ScramRuntimeException exception = assertThrows(ScramRuntimeException.class,
+      ScramInterruptedException exception = assertThrows(ScramInterruptedException.class,
           () -> CryptoUtil.hi(mac, password, salt, iterations));
 
-      assertTrue(exception.getMessage().contains("interrupted"),
+      assertTrue(exception.getMessage().contains("PBKDF2 computation interrupted"),
```
Comment (sehrope) [inline scram-common/src/test/java/com/ongres/scram/common/HiFunctionTest.java:133]: I don't think you even need this bit. Already checking for the specific exception class so the message itself doesn't matter.

**Review by sehrope — COMMENTED** [review]
Nice. This will make the pgjdbc testing easier too.

File: scram-common/src/test/java/com/ongres/scram/common/HiFunctionTest.java (hunk 1 above)
Comment (jorsol) [reply scram-common/src/test/java/com/ongres/scram/common/HiFunctionTest.java:47]: Yes, was the auto formatter, just ignore it.

File: scram-common/src/test/java/com/ongres/scram/common/HiFunctionTest.java (hunk 3 above)
Comment (jorsol) [reply scram-common/src/test/java/com/ongres/scram/common/HiFunctionTest.java:133]: I know, but still I don't mind if it is there.
